Automattic/wp-calypso
# Rollout plans redesign to all locales, desktop viewports only

#### Changes proposed in this Pull Request

* This is the global rollout PR for the new designs on the signup plans step. For context, read more on the experiment post pbxNRc-Hv-p2.
* Since the experiment was run for desktop viewports only, this PR launches the updated designs for desktop only. Mobile web will continue to retain its current design. 
* Most of the changes in the PR are towards removing unnecessary code. The `<PlanFeaturesComparison>` component introduced as part of this test is going to be used in signup flows only, so unlike the control version component `<PlanFeatures>` component, this new component does not have to worry about maintaining views for the Calypso /plans page. 

#### Testing instructions



* Go through the signup `onboarding` flow and confirm that the plans step matches one on one with the designs posted in the experiment post pbxNRc-Hv-p2. Try with both a free subdomain selection and custom domain selection(the first bullet point in the feature list will display the domain name for custom domains).
* For good measure, open production URL https://wordpress.com, assign yourself to treatment for this experiment from Abacus(experiment slug is `signup_plans_step_redesign_v1`, and verify that this PR's plans step page matches with treatment in production.
* Run through the site launch flow and confirm that the updated plans step UI is seen there
* Run through the signup flow in non-EN and confirm that translations are in place
* Run through the Add new site flow in non-EN and confirm that the treatment UI is present there.
* Verify that Tracks event are fired when selecting a paid plan or free plan.
* Confirm no changes to mobile web and that it's same as production
